File: cpp/class-object/complex.cpp
#include <iostream>

using namespace std;

class complex
{
private:
    float real;
    float imaginary;

public:
    void read()
    {
        cout << "Enter the real part of complex number: ";
        cin >> real;
        cout << "Enter the imaginary part of complex number: ";
        cin >> imaginary;
    }
    void display()
    {
        cout << "The number is " << real << " + " << imaginary <<"i" << endl;
    }
};

int main(void)
{
    //complex c;      //c is an object here (memory allocated statically)
    complex *c = new complex(); //c is pointer to object (memory allocated dynamically)
    c->read();
    c->display();
    return 0;
}

File: java/nopackage/classobject/Complex.java
import java.util.Scanner;

public class Complex {
    private float real;
    private float imaginary;

    public void read() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the real part of complex number: ");
        real = scanner.nextFloat();
        System.out.print("Enter the imaginary part of complex number: ");
        imaginary = scanner.nextFloat();
        scanner.close();
    }

    public void display() {
        System.out.println("The number is " + real + " + " + imaginary + "i");
    }

    public static void main(String[] args) {
        Complex complex = new Complex();    //here complex is a reference variable
        complex.read();
        complex.display();
    }
}
